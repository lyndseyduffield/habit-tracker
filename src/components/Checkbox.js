import React from "react";

const Checkbox = props => {
  return (
    <label class="checkbox-label">
      {props.children}
      {props.disabled ? (
        <span class="checkbox-custom checkbox-disabled rectangular"></span>
      ) : (
        <span class="checkbox-custom rectangular"></span>
      )}
    </label>
  );
};

export default Checkbox;
