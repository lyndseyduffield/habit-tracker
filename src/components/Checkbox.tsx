import React from "react";

interface Props {
  disabled: boolean;
}

const Checkbox: React.FC<Props> = (props) => {
  return (
    <label className="checkbox-label">
      {props.children}
      {props.disabled ? (
        <span className="checkbox-custom checkbox-disabled rectangular"></span>
      ) : (
        <span className="checkbox-custom rectangular"></span>
      )}
    </label>
  );
};

export default Checkbox;
