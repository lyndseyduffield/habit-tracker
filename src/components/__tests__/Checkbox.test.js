import React from "react";
import Checkbox from "../Checkbox";
import renderer from "react-test-renderer";

it("Renders disabled correctly", () => {
  const tree = renderer
    .create(
      <Checkbox disabled={true}>
        <span>My child</span>
      </Checkbox>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Renders enabled correctly", () => {
  const tree = renderer.create(<Checkbox disabled={false}></Checkbox>).toJSON();
  expect(tree).toMatchSnapshot();
});
