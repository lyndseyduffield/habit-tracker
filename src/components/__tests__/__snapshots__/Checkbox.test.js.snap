// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders disabled correctly 1`] = `
<label
  class="checkbox-label"
>
  <span>
    My child
  </span>
  <span
    class="checkbox-custom checkbox-disabled rectangular"
  />
</label>
`;

exports[`Renders enabled correctly 1`] = `
<label
  class="checkbox-label"
>
  <span
    class="checkbox-custom rectangular"
  />
</label>
`;
